android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java

```diff
@@ -121,18 +126,17 @@ protected void onResume() {
     }
 
     private void loadData() {
-        realmRepo.getSessionsByLocation(location)
-                .addChangeListener((sessions, orderedCollectionChangeSet) -> {
-                    this.sessions.clear();
-                    this.sessions.addAll(sessions);
-
-                    setUpcomingSession();
-                    sessionsListAdapter.setCopyOfSessions(sessions);
-                    sessionsListAdapter.notifyDataSetChanged();
-                    if (!Utils.isEmpty(searchText))
-                        sessionsListAdapter.filter(searchText);
-                    handleVisibility();
-                });
+        searchText=locationActivityViewModel.getSearchText();
+        locationActivityViewModel.getSessionByLocation(location).observe(this,sessionsList -> {
+            sessions.clear();
+            sessions.addAll(sessionsList);
+            setUpcomingSession();
+            sessionsListAdapter.setCopyOfSessions(sessions);
+            sessionsListAdapter.notifyDataSetChanged();
+            if (!Utils.isEmpty(searchText))
+                sessionsListAdapter.filter(searchText);
```

Comment:
Actually this filtering needs to be implemented inside the ```ViewModel```, not in the Adapter. Please refactor that also.